xarray/core/utils.py

```diff
@@ -123,9 +123,9 @@ def safe_cast_to_index(array: Any) -> pd.Index:
     """
     if isinstance(array, pd.Index):
         index = array
-    elif hasattr(array, "to_index"):
-        # xarray Variable
-        index = array.to_index()
+    elif hasattr(array, "_to_index"):
```

Comment:
Shall we make it an explicit `isinstance` check then? (same for Index and PandasIndexingAdapter below)

Reply:
Yes, makes sense. We should move `safe_cast_to_index` to another module too.